packages/side-nav/src/SideNav.spec.tsx

```diff
@@ -0,0 +1,224 @@
+  navEl?: renderedElement;
+  groupEl?: renderedElement;
+  headerEl?: renderedElement;
+  itemEl?: renderedElement;
+  linkEl?: renderedElement;
+}
+
+afterAll(cleanup);
+
+describe('packages/side-nav', () => {
+  const className = 'test-class-name';
+  const headerText = 'test-header-text';
+  const headerClassName = 'test-header-class-name';
+  const testIds = {
+    sideNav: 'side-nav',
+    sideNavGroup: 'side-nav-group',
+    sideNavItem: 'side-nav-item',
+    sideNavLink: 'side-nav-link',
+  };
+
+  let renderedEls: RenderedElements = {};
+
+  afterEach(() => {
+    document.body.innerHTML = '';
+    renderedEls = {};
+  });
+
+  describe('Side Nav Item', () => {
+    function renderSideNavItem(props: MenuItemProps = {}) {
+      const { sideNavItem } = testIds;
+      const { children, ...rest } = props;
+      const { getByTestId } = render(
+        <SideNavItem data-testid={sideNavItem} {...rest}>
+          {children}
+        </SideNavItem>,
+      );
+
+      renderedEls.itemEl = getByTestId(sideNavItem);
+    }
+
+    describe('when rendered with custom classes', () => {
+      beforeEach(() => {
+        renderSideNavItem();
+      });
+
+      test('it renders a menu item', () => {
+        expect(renderedEls.itemEl).toBeInTheDocument();
+        expect(renderedEls.itemEl?.tagName).toEqual('BUTTON');
+        expect(renderedEls.itemEl).toHaveAttribute(
+          'data-leafygreen-ui',
+          'menu-item-container',
+        );
+      });
+
+      test('it strips the expected menu item role from the parent', () => {
+        expect(renderedEls.itemEl?.parentNode).toHaveAttribute('role', 'none');
+      });
+    });
+
+    describe('when rendered as active', () => {
+      beforeEach(() => {
+        renderSideNavItem({ active: true });
+      });
+
+      test('it sets the aria attribute for the active item', () => {
+        expect(renderedEls.itemEl).toHaveAttribute('aria-current', 'page');
+      });
+    });
+
+    describe('when rendered as disabled', () => {
+      beforeEach(() => {
+        renderSideNavItem({ disabled: true });
+      });
+
+      test('it sets the aria attribute for the disabled item', () => {
+        expect(renderedEls.itemEl).toHaveAttribute('aria-disabled', 'true');
+      });
+    });
+
+    describe('when rendered as a link', () => {
+      beforeEach(() => {
+        renderSideNavItem({ href: '/v2#' });
+      });
+
+      test('it renders as an link menu item', () => {
+        expect(renderedEls.itemEl).toBeInTheDocument();
+        expect(renderedEls.itemEl?.tagName).toEqual('A');
+        expect(renderedEls.itemEl).toHaveAttribute('href', '/v2#');
+      });
+    });
+
+    describe('when rendered with children', () => {
+      beforeEach(() => {
+        const { sideNavLink } = testIds;
+        const children = (
+          <a href="#clusters" data-testid={sideNavLink}>
+            Clusters
+          </a>
+        );
+
+        renderSideNavItem({ children });
+      });
```

Comment:
[TODO] this lacks a test to detect the correctly rendered child (this was a brief oversight, it should be simple)